keys/keys.go

```diff
@@ -471,16 +471,27 @@ func MakeSplitKey(key roachpb.Key) (roachpb.Key, error) {
 // Range returns a key range encompassing all the keys in the Batch.
 // TODO(tschottdorf): there is no protection for doubly-local keys here;
 // maybe Range should return an error.
-func Range(ba roachpb.BatchRequest) roachpb.RSpan {
+func Range(ba roachpb.BatchRequest) (roachpb.RSpan, error) {
 	from := roachpb.RKeyMax
 	to := roachpb.RKeyMin
-	for _, arg := range ba.Requests {
+	for i, arg := range ba.Requests {
 		req := arg.GetInner()
 		if req.Method() == roachpb.Noop {
 			continue
 		}
 		h := req.Header()
 		key := Addr(h.Key)
+		if i == len(ba.Requests) {
```

Comment:
this condition is never true, and we also want to check each request, so just remove line 484 and 494. You're also not using `key`, so move that definition below your check.